lib/veri_traci.ml

```diff
@@ -0,0 +1,98 @@
+open Bap.Std
+open Bap_traces.Std
+
+module SM = Monad.State
+open SM.Monad_infix
+
+type 'a u = 'a Bil.Result.u
+type event = Trace.event
+
+let stub = fun _ -> SM.return () 
+
+class context trace =
+ object(self:'s)
+    inherit Bili.context
+    val events = Trace.events trace 
+    method next_event = match Seq.next events with
+      | None -> None 
+      | Some (ev,evs) -> Some ({<events = evs; >}, ev)
+
+    method add_events trace = {<events = Trace.events trace >}
```

Comment:
I think that the verb `add` is a bad choice here. "Add" assumes monotone extension of something. Here, you're actually resetting the environment and delete existing set of events. So you should either consider to rename is to `set_events`, or even better `with_events` (a usual convention for functional update function names); or you can consider to actually extend the set of events, by prepending or appending it to an existing sequence of events.